### PR TITLE
Implement stonecutters

### LINKED ISSUE #564: Implement stonecutters

### PR DESCRIPTION
**Pull request by JustTalDevelops**
Closes #564.

This PR also makes recipes require a string block name, for a couple of reasons:
- Stonecutters as a block go under the ID "stonecutter_block", but use "stonecutter" over the network.
- Blocks with properties such as facing directions will not be properly loaded on start, since we don't have any data about the possible properties of the block, and instead assume nil for the properties.

### CODE REVIEW

File: server/session/handler_stonecutter.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const stonecutterInputSlot = 0x03
+
+// handleStonecutting handles a CraftRecipe stack request action made using a stonecutter.
+func (h *ItemStackRequestHandler) handleStonecutting(a *protocol.CraftRecipeStackRequestAction, s *Session) error {
+	craft, ok := s.recipes[a.RecipeNetworkID]
+	if !ok {
+		return fmt.Errorf("recipe with network id %v does not exist", a.RecipeNetworkID)
+	}
+	if _, shapeless := craft.(recipe.Shapeless); !shapeless {
+		return fmt.Errorf("recipe with network id %v is not a shaped or shapeless recipe", a.RecipeNetworkID)
```
Comment (DaPigGuy) [inline server/session/handler_stonecutter.go:19]:
```suggestion
		return fmt.Errorf("recipe with network id %v is not a shapeless recipe", a.RecipeNetworkID)
```

**Review by DaPigGuy — APPROVED** [review]

**Review by Sandertv — APPROVED** [review]
This looks fine to me. Feel free to merge if it's been tested thoroughly.

**Comment by Sandertv** [comment]
And when the conflicts have been resolved.

**Comment by JustTalDevelops** [comment]
This should be ready now.

**Pull request merged**